docs/architecture/maui/mvvm-community-toolkit-features.md

```diff
@@ -0,0 +1,211 @@
+---
+title: MVVM Toolkit Features
+description: Helper components to simplify app development and maintenance
+author: michaelstonis
+no-loc: [MAUI]
+ms.date: 03/10/2023
+---
+
+# MVVM Toolkit Features
+
+[!INCLUDE [download-alert](includes/download-alert.md)]
+
+## MVVM Toolkit
+
+The [Model-View-ViewModel (MVVM)](mvvm.md) pattern is a great structural basis for creating our applications. In this pattern, the ViewModel becomes the backbone of our application as it provides communication to our front-end user interface and backing components. To provide integration with the user interface, we will rely on the ViewModel's properties and commands. As detailed in [Updating views in response to changes in the underlying view model or model](mvvm.md#updating-views-in-response-to-changes-in-the-underlying-view-model-or-model), the `INotifyPropertyChanged` interface on our ViewModel to allows changes to our properties to notify when the value is changed. Implementing all of these features means that our ViewModel can end up becoming very verbose. For example, the following code shows a simple ViewModel with properties that raise changes:
+
+```csharp
+public class SampleViewModel : INotifyPropertyChanged
+{
+    private string _name;
+    private int _value;
+
+    public event PropertyChangedEventHandler PropertyChanged;
+
+    public string Name
+    {
+        get => _name;
+        set => this.SetPropertyValue(ref _name, value);
+    }
+
+    public int Value
+    {
+        get => _value;
+        set => this.SetPropertyValue(ref _value, value);
+    }
+
+    protected bool SetPropertyValue<T>(ref T storageField, T newValue, [CallerMemberName] string propertyName = "")
```

Comment:
```suggestion
    protected bool SetPropertyValue<T>(
        ref T storageField,
        T newValue,
        [CallerMemberName] string propertyName = "")
```

I'm curious, why does this return `bool` at all? It's not used here, and it made me question its purpose.